Libraries/LibGfx/JPGLoader.cpp

```diff
@@ -0,0 +1,1057 @@
+    58, 59, 52, 45, 38, 31, 39, 46,
+    53, 60, 61, 54, 47, 55, 62, 63
+};
+
+using Marker = u16;
+
+void generate_huffman_codes(HuffmanTableSpec& table)
+{
+    unsigned code = 0;
+    for (auto number_of_codes : table.code_counts) {
+        for (int i = 0; i < number_of_codes; i++)
+            table.codes.append(code++);
+        code <<= 1;
+    }
+}
+
+Optional<size_t> read_huffman_bits(HuffmanStreamState& hstream, size_t count = 1)
+{
+    if (count > (8 * sizeof(size_t))) {
+        dbg() << String::format("Can't read %i bits at once!", count);
+        return {};
+    }
+    size_t value = 0;
+    while (count--) {
+        if (hstream.byte_offset >= hstream.stream.size()) {
+            dbg() << String::format("Huffman stream exhausted. This could be an error!");
+            return {};
+        }
+        u8 current_byte = hstream.stream[hstream.byte_offset];
+        u8 current_bit = 1u & (u32)(current_byte >> (7 - hstream.bit_offset)); // MSB first.
+        hstream.bit_offset++;
+        value = (value << 1) | (size_t)current_bit;
+        if (hstream.bit_offset == 8) {
+            hstream.byte_offset++;
+            hstream.bit_offset = 0;
+        }
+    }
+    return value;
+}
+
+Optional<u8> get_next_symbol(HuffmanStreamState& hstream, const HuffmanTableSpec& table)
+{
+    unsigned code = 0;
+    size_t code_cursor = 0;
+    for (int i = 0; i < 16; i++) { // Codes can't be longer than 16 bits.
+        auto result = read_huffman_bits(hstream);
+        if (!result.has_value())
+            return {};
+        code = (code << 1) | (i32)result.release_value();
+        for (int j = 0; j < table.code_counts[i]; j++) {
+            if (code == table.codes[code_cursor])
+                return table.symbols[code_cursor];
+            code_cursor++;
+        }
+    }
+
+    dbg() << "If you're seeing this...the jpeg decoder needs to support more kinds of JPEGs!";
+    return {};
+}
+
+/*
+     * Build the macroblocks possible by reading single (MCU) subsampled pair of CbCr.
+     * Depending on the sampling factors, we may not see triples of y, cb, cr in that
+     * order. If sample factors differ from one, we'll read more than one block of y-
+     * coefficients before we get to read a cb-cr block.
+     *
+     * In the function below, `hcursor` and `vcursor` denote the location of the block
+     * we're building in the macroblock matrix. `vfactor_i` and `hfactor_i` are cursors
+     * that iterate over the vertical and horizontal subsampling factors, respectively.
+     * When we finish one iteration of the innermost loop, we'll have the coefficients
+     * of one of the components of block at position `mb_index`. When the outermost loop
+     * finishes first iteration, we'll have all the luminance coefficients for all the
+     * macroblocks that share the chrominance data. Next two iterations (assuming that
+     * we are dealing with three components) will fill up the blocks with chroma data.
+     */
```

Comment:
I'm guessing `clang-format` didn't like that block comment, or just ignored it :)

There's very few multi-line block comments similar to this in the codebase, but back during [LibCrypto BigInts](https://github.com/SerenityOS/serenity/blob/master/Libraries/LibCrypto/BigInt/UnsignedBigInteger.cpp#L646), we ended up using the typedoc-like `/** <comment> */` syntax for function descriptions, and several `// <comment>`-style comments for big blocks in other places.